$OpenBSD: patch-src_nhttpd_proto_h,v 1.2 2011/03/17 17:44:23 jasper Exp $

Security fix for CVE-2011-0751
Nostromo Directory Traversal Vulnerability.

Patch from upstream CVS.

--- src/nhttpd/proto.h.orig	Tue Jun 23 16:33:23 2009
+++ src/nhttpd/proto.h	Thu Mar 17 18:40:37 2011
@@ -27,7 +27,9 @@ int		sdlisten(int, int);
 /*
  * http.c
  */
-int		http_verify(const char *, const char *, const int, const int);
+int		http_decode_header_uri(char *, const int);
+int		http_verify(char *, const int, const char *, const int,
+		    const int);
 int		http_proc(const char *, char *, const int, const int,
 		    const int);
 int		http_cgi_getexec(char *, char *, const char *, const int,
